tests/test_synced_collection.py

```diff
@@ -0,0 +1,587 @@
+        self.store(data)
+        assert synced_dict == data
+
+        # invalid data
+        data = [1, 2, 3]
+        with open(self._fn_, 'wb') as file:
+            file.write(json.dumps(data).encode())
+        with pytest.raises(ValueError):
+            synced_dict.load()
+
+    def test_copy_as_dict(self, synced_dict, testdata):
+        key = 'copy'
+        synced_dict[key] = testdata
+        copy = dict(synced_dict)
+        del synced_dict
+        assert key in copy
+        assert copy[key] == testdata
+
+    def test_nested_dict(self, synced_dict):
+        synced_dict['a'] = dict(a=dict())
+        child1 = synced_dict['a']
+        child2 = synced_dict['a']['a']
+        assert isinstance(child1, type(synced_dict))
+        assert isinstance(child1, type(child2))
+
+    def test_nested_dict_with_list(self, synced_dict):
+        synced_dict['a'] = [1, 2, 3]
+        child1 = synced_dict['a']
+        synced_dict['a'].append(dict(a=[1, 2, 3]))
+        child2 = synced_dict['a'][3]
+        child3 = synced_dict['a'][3]['a']
+        assert isinstance(child2, type(synced_dict))
+        assert isinstance(child1, type(child3))
+        assert isinstance(child1, SyncedCollection)
+        assert isinstance(child3, SyncedCollection)
+
+    def test_write_invalid_type(self, synced_dict, testdata):
+        class Foo(object):
+            pass
+
+        key = 'write_invalid_type'
+        synced_dict[key] = testdata
+        assert len(synced_dict) == 1
+        assert synced_dict[key] == testdata
+        d2 = Foo()
+        with pytest.raises(TypeError):
+            synced_dict[key + '2'] = d2
+        assert len(synced_dict) == 1
+        assert synced_dict[key] == testdata
+
+    def test_keys_with_dots(self, synced_dict):
+        with pytest.raises(InvalidKeyError):
+            synced_dict['a.b'] = None
+
+    def test_keys_valid_type(self, synced_dict, testdata):
+
+        class MyStr(str):
+            pass
+        for key in ('key', MyStr('key'), 0, None, True):
+            synced_dict[key] = testdata
+            assert str(key) in synced_dict
+            assert synced_dict[str(key)] == testdata
+
+    def test_keys_invalid_type(self, synced_dict, testdata):
+
+        class A:
+            pass
+        for key in (0.0, A(), (1, 2, 3)):
+            with pytest.raises(KeyTypeError):
+                synced_dict[key] = testdata
+        for key in ([], {}, dict()):
+            with pytest.raises(TypeError):
+                synced_dict[key] = testdata
+
+
+class TestJSONList:
+
+    _write_concern = False
+
+    @pytest.fixture(autouse=True)
+    def synced_list(self):
+        self._tmp_dir = TemporaryDirectory(prefix='jsondict_')
+        self._fn_ = os.path.join(self._tmp_dir.name, FN_JSON)
+        self._cls = JSONList
+        self._backend_kwargs = {'filename': self._fn_, 'write_concern': self._write_concern}
+        yield self._cls(**self._backend_kwargs)
+        self._tmp_dir.cleanup()
```

Comment:
Same issue as with the other test class: please remove the use of the implicit class variable.